recipes/armadillo/all/conanfile.py

```diff
@@ -271,10 +306,25 @@ def package(self):
         copy(self, "NOTICE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "share"))
+        self._create_cmake_module_variables(os.path.join(self.package_folder, self._module_vars_rel_path))
+
 
     def package_info(self):
         self.cpp_info.libs = ["armadillo"]
         self.cpp_info.set_property("pkg_config_name", "armadillo")
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "Armadillo")
+        self.cpp_info.set_property("cmake_target_name", "Armadillo::Armadillo")
+        self.cpp_info.set_property("cmake_target_aliases", ["Armadillo", "armadillo", "armadillo::armadillo", "Armadillo::armadillo"])
```

Comment:
If you add alias, I would limit them to existing case then: 

```suggestion
        self.cpp_info.set_property("cmake_target_aliases", ["armadillo", "armadillo::armadillo"])
```